timemachine/fe/utils.py

```diff
@@ -241,6 +241,13 @@ def sanitize_energies(full_us, lamb_idx, cutoff=10000):
     return np.where(abs_us < cutoff, full_us, np.inf)
 
 
+def read_sdf(fname: str) -> List[Chem.Mol]:
+    """Read list of mols from an SDF (without discarding hydrogens!)"""
+    supplier = Chem.SDMolSupplier(str(fname), removeHs=False)
+    mols = [mol for mol in supplier]
+    return mols
```

Comment:
I don't see a strong reason either way -- just meant to be simplest / most convenient for the most typical use.

Initially picked list since it felt marginally simpler (and anticipated usage is never to read files large enough for wasted molecule-reading effort to be noticeable).

Other context: Most usages in `timemachine/tests` load the whole file as a list, although we do use the additional flexibility of a generator a few times, e.g. to read only the first molecule in an SDF. (Usage counts here: https://github.com/proteneer/timemachine/pull/892/commits/67c77dc304be8c5a23f181b4488b10e9d230b1c3 )

Reply:
Makes sense to me to leave it as is if we don't foresee any use for the generator interface.